services/inputhost/pathCache.go

```diff
@@ -508,3 +536,26 @@ func (pathCache *inPathCache) getState() *admin.DestinationState {
 
 	return destState
 }
+
+func (pathCache *inPathCache) drainExtent(extUUID string, updateUUID string, drainWG *sync.WaitGroup) {
+	defer drainWG.Done()
+	pathCache.RLock()
+	defer pathCache.RUnlock()
+
+	extCache, ok := pathCache.extentCache[extentUUID(extUUID)]
+	if !ok {
+		// draining called for extent which is not present
+		// just return
+		return
+	}
+
+	if extCache.connection.prepDrain() {
+		var connDrainWG sync.WaitGroup
+		// first send DRAIN command to all connections
+		// then start draining extents
+		pathCache.drainConnections(updateUUID, &connDrainWG)
+		// do best effort waiting to notify all connections here
+		common.AwaitWaitGroup(&connDrainWG, connWGTimeout)
+		extCache.connection.drain()
```

Comment:
remind me, why is this simply not extCache.connection.close()  ? At this point, we have waited enough for the connections to drain, all we have to do is unload everything, right ?